### PR TITLE
feat: show team members in access dialog

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

### <samp>🤖 Generated by Copilot at 45dfb61</samp>

This pull request adds a new feature to the journeys-admin app that allows the user to see the team members of a journey in the access dialog component. It updates the query, types, components, stories, and tests to support this feature. It also deletes an unused file and adds some translation and debugging features.

[- Link to Basecamp Todo
](https://3.basecamp.com/3105655/buckets/33184123/todos/6444395387)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] it show not show empty access dialog in main page
- [ ] it should show team members in access dialog 
- [ ] team members section in access dialog is read only
- [ ] form does not validate against read only team members since team members permissions are different to editor permissions
- [ ] team members cannot be seen in "Shared With Me" page
- [ ] team member invitations not present in access dialog list

# Walkthrough

### <samp>🤖 Generated by Copilot at 45dfb61</samp>

*  Add a new query and types for fetching a journey with its team data ([link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-f4c0310ff2f03e9bd78b9246b885d6680a55017bfd6bed1404c3d0802e367c38R1-R63))
*  Update the AccessDialog component to use the new query and types, and to render the team members list using the UserTeamList component ([link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-50ca6c8c898865f557fcb72ef386044843d1fb39d4294e730b5f60851414a6ebL4-R41), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-50ca6c8c898865f557fcb72ef386044843d1fb39d4294e730b5f60851414a6ebL62-R87), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-50ca6c8c898865f557fcb72ef386044843d1fb39d4294e730b5f60851414a6ebR137), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-50ca6c8c898865f557fcb72ef386044843d1fb39d4294e730b5f60851414a6ebR152-R160))
*  Update the test and story files for the AccessDialog component to use the new query and types, and to provide mock data for the team members ([link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-ece6cd44f7d23895a5424da507167e3eda3a4bcb7290844f585a62ca97b510dfL7-R7), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-ece6cd44f7d23895a5424da507167e3eda3a4bcb7290844f585a62ca97b510dfL41-R41), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-ece6cd44f7d23895a5424da507167e3eda3a4bcb7290844f585a62ca97b510dfR50-R95), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-ece6cd44f7d23895a5424da507167e3eda3a4bcb7290844f585a62ca97b510dfR181-R217), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-31339a9406c51b321bd2a7fda289cdd598fc6487b06f4aa92b224c56aaac4e01L10-R10), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-31339a9406c51b321bd2a7fda289cdd598fc6487b06f4aa92b224c56aaac4e01L29-R29), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-31339a9406c51b321bd2a7fda289cdd598fc6487b06f4aa92b224c56aaac4e01R38-R83))
*  Update the UserTeamList component to accept a variant prop that can be either 'readonly' or 'default', and to render the team members differently depending on the variant ([link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-1198230c863e942788c29e8d8c79f6f80872c102a2cee7490c85c2b42bd1952aR10), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-1198230c863e942788c29e8d8c79f6f80872c102a2cee7490c85c2b42bd1952aL19-R26), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-1198230c863e942788c29e8d8c79f6f80872c102a2cee7490c85c2b42bd1952aL27-R37), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-1198230c863e942788c29e8d8c79f6f80872c102a2cee7490c85c2b42bd1952aL35-R47), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-1198230c863e942788c29e8d8c79f6f80872c102a2cee7490c85c2b42bd1952aL72-R84))
*  Update the UserTeamListItem component to accept a variant prop and to hide or disable the buttons if the variant is readonly ([link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-332992b727d8f3c48cac5c4ffb075a69d286a0f6272584381a4d39e28cc90feaR26), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-332992b727d8f3c48cac5c4ffb075a69d286a0f6272584381a4d39e28cc90feaL41-R43), [link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-332992b727d8f3c48cac5c4ffb075a69d286a0f6272584381a4d39e28cc90feaL96-R107))
*  Add a test case for the UserTeamList component to check if the buttons are disabled if the variant is readonly ([link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-40b30375e293c176beec728bf09a1bb5973b9da5810e9a1bf9993f1714bc5757R134-R148))
*  Delete the old query and types for fetching a journey with its user journeys ([link](https://github.com/JesusFilm/core/pull/1822/files?diff=unified&w=0#diff-3f8e2faae10ee96305e35a4bb2db1d2407b2f928d6b14c2df98a450a71adf5f3))

### CODE REVIEW

File: apps/journeys-admin/src/components/AccessDialog/AccessDialog.tsx [hunk 1]
```diff
@@ -112,6 +134,7 @@ export function AccessDialog({
     }
   }, [open, refetch, refetchInvites, loadUser])
 
+  console.log(data?.journey.team)
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/AccessDialog/AccessDialog.tsx:137]: sadge, remove

File: apps/journeys-admin/src/components/Team/TeamManageDialog/UserTeamList/UserTeamList.tsx [hunk 2]
```diff
@@ -16,27 +17,34 @@ import { UserTeamRole } from '../../../../../__generated__/globalTypes'
...
 }
 
 export function UserTeamList({
   data,
   currentUserTeam,
-  loading
+  loading,
+  variant = 'default'
 }: UserTeamListProps): ReactElement {
+  console.log(data)
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Team/TeamManageDialog/UserTeamList/UserTeamList.tsx:35]: more sadge, remove

File: apps/journeys-admin/src/components/AccessDialog/AccessDialog.tsx [hunk 3]
```diff
@@ -1,26 +1,44 @@
...
 import { UserJourneyRole } from '../../../__generated__/globalTypes'
 import { useCurrentUser } from '../../libs/useCurrentUser'
+import { UserTeamList } from '../Team/TeamManageDialog/UserTeamList'
 
 import { AddUserSection } from './AddUserSection'
 import { UserList } from './UserList'
 
-export const GET_JOURNEY_WITH_USER_JOURNEYS = gql`
-  query GetJourneyWithUserJourneys($id: ID!) {
+export const GET_JOURNEY_WITH_USER_JOURNEYS_AND_USER_TEAMS = gql`
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/AccessDialog/AccessDialog.tsx:24]: NIT: name kinda long aye, up to you but somehow try make smaller

Comment (Kneesal) [reply apps/journeys-admin/src/components/AccessDialog/AccessDialog.tsx:24]: will see what Tatai says in his review

Comment (tataihono) [reply apps/journeys-admin/src/components/AccessDialog/AccessDialog.tsx:24]: name kinda long aye

Comment (tataihono) [reply apps/journeys-admin/src/components/AccessDialog/AccessDialog.tsx:24]: could go with GET_JOURNEY_WITH_PERMISSIONS

**Review by tataihono — CHANGES_REQUESTED** [review]
I think we need to do some permissions work. You should not be able to list all the members of a team if you are not a member of that team.

**Review by GeronimoJohn — DISMISSED** [review]

**Review by tataihono — DISMISSED** [review]

File: apps/journeys-admin/src/components/AccessDialog/AddUserSection/AddUserSection.tsx [hunk 4]
```diff
@@ -43,15 +43,31 @@ export function AddUserSection({
   return (
     <Stack flexGrow={1} sx={{ m: 4, mt: 2 }}>
       <Stack direction="row" alignItems="center" sx={{ mb: 4 }}>
-        <GroupAddIcon />
+        <GroupAddIcon sx={{ transform: 'scaleX(-1)' }} />
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/AccessDialog/AddUserSection/AddUserSection.tsx:46]: Hey sorry, but icon conversations seems to just catch you all the time hahaha. 

Chatted with Ziwei about this one, a few things we've noticed:
- The materila ui icon and the ones from the design are different. (not referring it being flipped - see photos)
Screenshot 2023-08-24 at 12 04 27 PM
Screenshot 2023-08-24 at 12 04 22 PM
The bottom is more rounded and there's more space between the two avatars. I like the idea behind it and you it as close to the design. But the concern that we have is that in a way it's like they're introducing a new icon here, can't seem to find it in MUI nor Stratis
- it seems like we're not using how MUI have intended it for use

File: apps/journeys-admin/src/components/AccessDialog/AddUserSection/AddUserSection.tsx [hunk 5]
```diff
@@ -43,15 +43,31 @@ export function AddUserSection({
...
         <Typography variant="subtitle1" sx={{ marginLeft: 3 }}>
           {t('Add editor by')}
         </Typography>
         <Button
           variant="outlined"
           size="small"
           startIcon={
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/AccessDialog/AddUserSection/AddUserSection.tsx:53]: NIT: if you want a little bit more 'challenge' or 'push' within this button component. How can you stop repeating the icon styling

File: apps/journeys-admin/src/components/AccessDialog/UserList/UserListItem/PromoteUser/PromoteUser.tsx [hunk 6]
```diff
@@ -27,6 +29,7 @@ export const USER_JOURNEY_PROMOTE = gql`
 `
 
 export function PromoteUser({ id, onClick }: PromoteUserProps): ReactElement {
+  const { t } = useTranslation('apps-journeys-admin')
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/AccessDialog/UserList/UserListItem/PromoteUser/PromoteUser.tsx:32]: useTranslations sometimes gives this warning within our test logs. Which can be annoying if you're testing and trying to find errors. Can avoid via mocking useTranslations. There should be examples within Core. 

Screenshot 2023-08-24 at 12 12 53 PM

Gonna have to mock it for all the files where you've used useTranslations

**Review by GeronimoJohn — APPROVED** [review]

**Pull request merged**